src/OpenDebugAD7/TracepointManager.cs

```diff
@@ -0,0 +1,229 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.VisualStudio.Debugger.Interop;
+using Microsoft.VisualStudio.Debugger.Interop.DAP;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text.RegularExpressions;
+
+namespace OpenDebugAD7
+{
+    internal class TracepointManager
+    {
+        private readonly object _lock = new object();
+
+        private readonly Regex regex;
+        private readonly Dictionary<uint, string> m_breakpointLogMessages;
+        private readonly enum_DEBUGPROP_INFO_FLAGS propertyInfoFlags;
+        private readonly enum_EVALFLAGS flags;
+
+        internal TracepointManager()
+        {
+            m_breakpointLogMessages = new Dictionary<uint, string>();
+
+            // Matches strings that are in { } 
+            // or upper case strings that begin with $.
+            regex = new Regex("\\$[A-Z]+|{.*?}", RegexOptions.Compiled);
```

Comment:
>regex = new Regex("\\$[A-Z]+|{.*?}", RegexOptions.Compiled); [](start = 12, length = 60)

I am skeptical that you want to use Regex. Here are some test cases to try:

`{wcscmp(L"{hi}", argv[0])}`
`\{Escape Test\}`